### PR TITLE
COM-292: Refactor theming of `AppHeaderDropdown` 

### PR DESCRIPTION
**Pull request by andrearutrecht**
COM-292

Styling is refactored for `AppHeaderDropdown`.

<details><summary>Demo Video of `AppHeaderDropdown`</summary>
<p>

https://github.com/vivid-planet/comet/assets/122883866/b5e2b3d4-1374-459c-813e-8c960490a1f9

</p>
</details> 

- [x] Add changeset (if necessary)

### CODE REVIEW

File: packages/admin/admin/src/appHeader/dropdown/AppHeaderDropdown.tsx [hunk 1]
```diff
@@ -16,17 +22,25 @@ export interface AppHeaderDropdownProps extends Omit<AppHeaderButtonProps, "chil
...
+    },
+})(
+    css`
+        height: 100%;
+    `,
+);
+
+const PopoverRoot = styled(Popover, {
+    name: "CometAdminAppHeaderDropdown",
+    slot: "popoverRoot",
```
Comment (johnnyomair) [inline packages/admin/admin/src/appHeader/dropdown/AppHeaderDropdown.tsx:39]: I'd call this just `popover`. @jamesricky what do you think? IMO is a breaking change preferable to an inferior API. Also, we're already removing another element (paper)

Comment (jamesricky) [reply packages/admin/admin/src/appHeader/dropdown/AppHeaderDropdown.tsx:39]: Yes probably makes sense.

File: packages/admin/admin/src/appHeader/dropdown/AppHeaderDropdown.tsx [hunk 2]
```diff
@@ -68,41 +85,43 @@ function Dropdown({
...
     }
 
+    interface ComponentNameToClassKey {
+        CometAdminAppHeaderDropdown: AppHeaderDropdownClassKey;
+    }
+
     interface Components {
         CometAdminAppHeaderDropdown?: {
-            defaultProps?: ComponentsPropsList["CometAdminAppHeaderDropdown"];
+            defaultProps?: Partial<ComponentsPropsList["CometAdminAppHeaderDropdown"]>;
```
Comment (jamesricky) [inline packages/admin/admin/src/appHeader/dropdown/AppHeaderDropdown.tsx:124]: Only one of the prop-types (`ComponentsPropsList` or `Components`)  needs to be `Partial`, as one is used by the other.

File: packages/admin/admin/src/appHeader/dropdown/AppHeaderDropdown.tsx [hunk 3]
```diff
@@ -38,16 +52,19 @@ function DefaultArrowDown(): React.ReactElement {
...
-}: AppHeaderDropdownProps & WithStyles<typeof styles>): React.ReactElement {
+export function AppHeaderDropdown(inProps: AppHeaderDropdownProps) {
+    const {
+        children,
+        buttonChildren,
+        dropdownArrow = (isOpen: boolean) => (isOpen ? <DefaultArrowUp /> : <DefaultArrowDown />),
+        popoverProps,
+        open,
+        onOpenChange,
+        classes,
```
Comment (jamesricky) [inline packages/admin/admin/src/appHeader/dropdown/AppHeaderDropdown.tsx:63]: `classes` should no longer exist or be used. 

```suggestion
```

**Review by jamesricky — CHANGES_REQUESTED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**